CMakeLists_files.cmake

```diff
@@ -574,9 +574,40 @@ list (APPEND TEST_DATA_FILES
 # originally generated with the command:
 # find opm -name '*.h*' -a ! -name '*-pch.hpp' -printf '\t%p\n' | sort
 list (APPEND PUBLIC_HEADER_FILES
+  flow/flow_blackoil.hpp
```

Comment:
these (mostly) contain prototypes for functions that are never installed, ie not in the library. if we need need these in downstreams there is more work to be done, this achieves nothing.

Reply:
yes, i am aware of that but Main.hpp is also not structured to be installed, it's only meant to end up in simulators, which again are not meant to be downstream accessible. 

my point is that if we really want to prepare the code for downstreams we have a lot of things that need to be addressed, not just in the code but also in the build system., e.g. how we handle gpu stuff.

i'm not really opposing this change, it makes thing no worse than it already was and helps a little so...

Reply:
> yes, i am aware of that but Main.hpp is also not structured to be installed, it's only meant to end up in simulators, which again are not meant to be downstream accessible.

True.  I guess it slipped through in #2532 (8c2ba0bfb6) then.

Reply:
> my point is that if we really want to prepare the code for downstreams we have a lot of things that need to be addressed, not just in the code but also in the build system., e.g. how we handle gpu stuff.

I think we may getting to a point where a careful review of the full OPM build system is in order anyway.